github/github.go

```diff
@@ -107,6 +107,7 @@ type Client struct {
 	rateMu     sync.Mutex
 	rateLimits [categories]Rate // Rate limits for the client as determined by the most recent API calls.
 	mostRecent rateLimitCategory
+	common     service
```

Comment:
I have a style suggestion.

```Go
rateMu     sync.Mutex
rateLimits [categories]Rate
mostRecent rateLimitCategory
```

Here, `rateMu` is a mutex hat. Like a hat, it sits on top of the variables that it protects.

So, **without needing to actually write the comment**, the above is implicitly understood to be equivalent to:

```Go
// rateMu protects rateLimits and mostRecent fields.
rateMu     sync.Mutex
rateLimits [categories]Rate
mostRecent rateLimitCategory
```

When adding a new, unrelated field that isn't protected by `rateMu`, don't do this:

```diff
 rateMu     sync.Mutex
 rateLimits [categories]Rate
 mostRecent rateLimitCategory
+common     service
```

Instead, do this:

```diff
 rateMu     sync.Mutex
 rateLimits [categories]Rate
 mostRecent rateLimitCategory
+
+common     service
```

How does that sound?

I can't remember where I first saw this pattern, but it might've been @davecheney. I really like it and recommend not breaking it. (I'll try to fix up the `clientMu` mutex on top to use this pattern too.)